ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py

```diff
@@ -106,41 +106,41 @@ def add_verify_peers_test(description: str, afi: str, bgp_neighbor_ip: str, safi
+            anta_tests.setdefault("generic", []).append(
+                {
+                    "VerifyRoutingProtocolModel": {
+                        "model": "multi-agent",
+                        "result_overwrite": {"categories": self.categories, "description": "ArBGP is configured and operating", "custom_field": "ArBGP"},
+                    }
                 }
-            }
-        )
-
-        bgp_peer_groups = get(self.hostvars[self.device_name], "router_bgp.peer_groups", [])
-
-        for bgp_neighbor in get(self.hostvars[self.device_name], "router_bgp.neighbors", []):
-            # TODO - this matches legacy eos_validate_state BUT works only for neighbors in peer groups...
-            try:
-                neighbor_peer_group = get_item(
-                    bgp_peer_groups, "name", bgp_neighbor["peer_group"], required=True, var_name=f"name: {bgp_neighbor['peer_group']}"
-                )
-                bgp_neighbor_ip = str(get(bgp_neighbor, "ip_address", required=True))
-
-                if get(neighbor_peer_group, "type", required=True) == "ipv4":
-                    add_verify_peers_test(description="ip bgp peer state established (ipv4)", afi="ipv4", safi="unicast", bgp_neighbor_ip=bgp_neighbor_ip)
-                elif get(neighbor_peer_group, "type", required=True) == "evpn":
-                    add_verify_peers_test(description="bgp evpn peer state established (evpn)", afi="evpn", bgp_neighbor_ip=bgp_neighbor_ip)
-
-            except AristaAvdMissingVariableError as e:
-                LOGGER.warning("Variable '%s' is missing. Please validate the Router BGP data model of this host.", str(e))
+            )
+
+            bgp_peer_groups = get(self.hostvars[self.device_name], "router_bgp.peer_groups", [])
+
+            for bgp_neighbor in get(self.hostvars[self.device_name], "router_bgp.neighbors", []):
+                # TODO - this matches legacy eos_validate_state BUT works only for neighbors in peer groups...
+                try:
+                    neighbor_peer_group = get_item(
+                        bgp_peer_groups, "name", bgp_neighbor["peer_group"], required=True, var_name=f"name: {bgp_neighbor['peer_group']}"
+                    )
+                    bgp_neighbor_ip = str(get(bgp_neighbor, "ip_address", required=True))
+
+                    if get(neighbor_peer_group, "type", required=True) == "ipv4":
+                        add_verify_peers_test(description="ip bgp peer state established (ipv4)", afi="ipv4", safi="unicast", bgp_neighbor_ip=bgp_neighbor_ip)
+                    elif get(neighbor_peer_group, "type", required=True) == "evpn":
+                        add_verify_peers_test(description="bgp evpn peer state established (evpn)", afi="evpn", bgp_neighbor_ip=bgp_neighbor_ip)
+
+                except AristaAvdMissingVariableError as e:
+                    LOGGER.warning("Variable '%s' is missing. Please validate the Router BGP data model of this host.", str(e))
```

Comment:
This error message is a bit misleading since it fails if the neighbor has no peer group or if peer group type is not set. BTW that peer group "type" should not be used, since a peer group can be part of multiple address families - that can be fixed later.

Reply:
Logs fixed in 650a0ce7ba7ab114c55fa46edb761b7716d59212